lean/main/dsls.lean

```diff
@@ -231,7 +231,8 @@ will be invoked in succession:
 -/
 
 def eg_AExp_fail: AExp :=
-  [imp_aexp'| 42 + 43]
+  [imp_aexp'| 42 + 43]  -- something goes wrong in the Infoview?  I see the commented text
```

Comment:
There are two `elab`'s trying to take over this one. That's why the double errors.

@bollu can you help us here?

Reply:
@arthurpaulino I don't understand. For me, the view looks as follows:

![image](https://user-images.githubusercontent.com/1694861/171639014-a848f55b-636f-4a26-8a80-73add797c44e.png)

At line 233, I have the expected error:

```
dsls.lean:233:0
elab_aexp_ident failed.
elab_AExp_num failed
All Messages (33)
```

And further on, at line `272`, I have the errors:

```
dsls.lean:272:0
elab_AExp_num failed
elab_AExp_num failed
elab_aexp_ident failed.
elab_AExp_num failed
```

So I don't  see double errors on line `233`.

#### Full transcript of my Infoview:

```
dsls.lean:233:15
Expected type
⊢ AExp
Messages (1)
dsls.lean:233:0
elab_aexp_ident failed.
elab_AExp_num failed
All Messages (33)
dsls.lean:101:0
AExp.ANat 42
dsls.lean:156:0
AExp.ANat 42
dsls.lean:172:0
AExp.AVar "foo"
dsls.lean:195:0
elab_AExp_num failed
dsls.lean:198:0
AExp.AVar "foo"
dsls.lean:205:0
elab_aexp_ident failed.
dsls.lean:207:0
AExp.ANat 43
dsls.lean:233:0
elab_aexp_ident failed.
elab_AExp_num failed
dsls.lean:234:2
overloaded, errors 
  unexpected syntax
    [imp_aexp'|42+43]
      -- elab_aexp_ident failed.
      -- elab_aexp_num failed
      
      /-
      Clearly, both parses `elab_aexp_ident` and `elab_aexp_num`
      are tried in succession, and both fail, leading to an error.
      
      We shall fix this omission, and finally parse the addition node.
      We go directly to parsing with `elab`, since the `macro_rules`
      approach does not have anything more interesting to say.
      -/
  
  unexpected syntax
    [imp_aexp'|42+43]
      -- elab_aexp_ident failed.
      -- elab_aexp_num failed
      
      /-
      Clearly, both parses `elab_aexp_ident` and `elab_aexp_num`
      are tried in succession, and both fail, leading to an error.
      
      We shall fix this omission, and finally parse the addition node.
      We go directly to parsing with `elab`, since the `macro_rules`
      approach does not have anything more interesting to say.
      -/
dsls.lean:272:0
elab_AExp_num failed
elab_AExp_num failed
elab_aexp_ident failed.
elab_AExp_num failed

```

Reply:
What I find strange is the repeated
```
/-
      Clearly, both parses `elab_aexp_ident` and `elab_aexp_num`
      are tried in succession, and both fail, leading to an error.
      
      We shall fix this omission, and finally parse the addition node.
      We go directly to parsing with `elab`, since the `macro_rules`
      approach does not have anything more interesting to say.
      -/
```

Reply:
Let me be more explicit: it seems normal to get two errors.  However, the two errors should simply print 
```
elab_aexp_ident failed.
elab_aexp_num failed
```
(and possibly an `overloaded` something).  However, each one of these errors seems to also collect the subsequent comment block:

```
/-
Clearly, both parses `elab_aexp_ident` and `elab_aexp_num`
are tried in succession, and both fail, leading to an error.

We shall fix this omission, and finally parse the addition node.
We go directly to parsing with `elab`, since the `macro_rules`
approach does not have anything more interesting to say.
-/
```